docs/about/release_process.rst

```diff
@@ -34,22 +34,26 @@ stable version.
 #. Merge changes leading up to the release into the `stable` branch
     This will also update the `stable` docs
 
-#. Upload the build to PyPi.
+#. Upload the build to PyPi
+    You might need a PyPI_ account with appropriate authorization.
```

Comment:
We should mention that the project is `datacube`, as a new dev might look for `datacube-core` or `opendatacube`
https://pypi.python.org/pypi/datacube/